lib/kitchen/directions/bake_figure.rb

```diff
@@ -5,29 +5,48 @@ module Directions
     module BakeFigure
       def self.v1(figure:, number:, cases: false)
         warn 'warning! exclude unnumbered figures from `BakeFigure` loop' if figure.unnumbered?
-        figure.wrap(%(<div class="os-figure#{' has-splash' if figure.has_class?('splash')}">))
+        figure.wrap(%(<div class="os-figure#{if figure.has_class?('splash')
+                                               ' has-splash'
+                                             elsif figure.has_class?('mechanism-figure')
+                                               ' has-mechanism-figure'
+                                             end}">))
 
         # Store label information
         figure.target_label(label_text: 'figure', custom_content: number, cases: cases)
-
         title = figure.title&.cut
         caption = figure.caption&.cut
         caption&.name = 'span'
         caption&.add_class('os-caption')
         caption&.remove_attribute('id') # only needed for pl books, where captions have id
 
-        figure.append(sibling:
-          <<~HTML.chomp
-            <div class="os-caption-container">
-              <span class="os-title-label">#{I18n.t("figure#{'.nominative' if cases}")} </span>
-              <span class="os-number">#{number}</span>
-              #{"<span class=\'os-divider\'> </span>" if title}
-              #{"<span class=\'os-title\' data-type=\'title\' id=\"#{title.id}\">#{title.children}</span>" if title}
-              #{"<span class=\'os-divider\'> </span>" if caption}
-              #{caption}
-            </div>
-          HTML
-        )
+        if figure.has_class?('mechanism-figure')
```

Comment:
this class seems like it's probably specific to ochem. i wonder if it would be possible to make this more general. for example, pass in a `prepend_caption_class` as a variable/`option` on `BakeFigure`, and then send in `'mechanism-figure'` from the recipe

Reply:
@kswanson33 I have refactored the code to accept `custom_class` variable with value passed by recipe. Let me know if this is what you have in mind :slightly_smiling_face: